src/rqt_plot/plot_widget.py

```diff
@@ -39,32 +40,31 @@
 from python_qt_binding.QtGui import QIcon
 from python_qt_binding.QtWidgets import QAction, QMenu, QWidget
 
-from rqt_py_common.topic_completer import TopicCompleter
-from rqt_py_common import topic_helpers, message_helpers
-
-from rqt_plot.rosplot import ROSData, RosPlotException
+from rosidl_parser.definition import AbstractGenericString
+from rosidl_parser.definition import AbstractNestedType
+from rosidl_parser.definition import AbstractSequence
+from rosidl_parser.definition import Array
+from rosidl_parser.definition import BasicType
+from rosidl_parser.definition import BOOLEAN_TYPE
+from rosidl_parser.definition import NamespacedType
 
+from rosidl_runtime_py.utilities import get_message
+from rosidl_runtime_py.utilities import get_message_namespaced_type
+from rosidl_runtime_py import import_message_from_namespaced_type
 
-def _parse_type(topic_type_str):
-    slot_type = topic_type_str
-    is_array = False
-    array_size = None
+from rqt_py_common.topic_completer import TopicCompleter
 
-    array_idx = topic_type_str.find('[')
-    if array_idx < 0:
-        return slot_type, False, None
+from rqt_plot.rosplot import ROSData, RosPlotException
 
-    end_array_idx = topic_type_str.find(']', array_idx + 1)
-    if end_array_idx < 0:
-        return None, False, None
+ARRAY_TYPE_REGEX = re.compile(r'(.*)\[(.*)\]')
```

Comment:
Is zero leading characters valid (e.g. `[2]`) ?

```suggestion
ARRAY_TYPE_REGEX = re.compile(r'(.+)\[(.*)\]')
```

Reply:
It doesn't matter much because it will fail [here](https://github.com/ros-visualization/rqt_plot/blob/26b5c3d9d9c1dd6f2b5d52c09500ab95954f20d6/src/rqt_plot/plot_widget.py#L117) anyway, but sounds good

Reply:
See https://github.com/ros-visualization/rqt_plot/pull/71/commits/b0e08b33e593139c0d177d1f4ab516caab630d07